src/main/java/duke/entities/Task.java

```diff
@@ -0,0 +1,145 @@
+package duke.entities;
+
+import java.time.LocalDate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.enums.TaskType;
+import duke.exceptions.DukeException;
+import duke.utils.CustomValidator;
+
+/**
+ * Represents a Task Object.
+ */
+public abstract class Task {
+    /** Task validation **/
+    public static final Pattern EVENT =
+            Pattern.compile("^(event) (?<description>.+) /from (?<from>.+) /to (?<to>.+)$",
+                    Pattern.CASE_INSENSITIVE);
+    public static final Pattern DEADLINE =
+            Pattern.compile("^(deadline) (?<description>.+) /by (?<by>.+)$",
+                    Pattern.CASE_INSENSITIVE);
+    public static final Pattern TODO = Pattern.compile("^(todo) (?<description>.+)$", Pattern.CASE_INSENSITIVE);
```

Comment:
Perhaps can put a line break to follow the format of the code in the lines before.